real_preprocessing/src/corner_detection.cpp

```diff
@@ -0,0 +1,88 @@
+#include "ros/ros.h"
+#include "apriltag_ros/AprilTagDetectionArray.h"
+#include "sstream"
+#include "tf_conversions/tf_eigen.h"
+#include "Eigen/Dense"
+#include "fstream"
+#include "string"
+
+void tfCallback(const apriltag_ros::AprilTagDetectionArray::ConstPtr& msg)
+{
+  Eigen::MatrixXd intrinsic(3, 3);  // potential calibration function
+  intrinsic << 630.3467672874693, 0, 314.10832937485145, 0, 624.9044125522948, 241.59156511756711, 0, 0, 1;
+  if (msg->detections.empty())
+  {
+    // no detections
+  }
+  else
+  {
+    std::string filename("detections_0000.yml");  // output path
+    std::ofstream fout;
+    fout.open(filename.c_str());
+    fout << "detections:";
+
+    for (float i = (msg->detections.size() - 1); i > -1; i--)
+    {
+      // reading data from topic
+      float xpos = msg->detections[i].pose.pose.pose.position.x;
```

Comment:
These names are not very descriptive and too generalized. They should ideally have the parent and child frames associated with them.